src/main/java/seedu/address/model/person/Name.java

```diff
@@ -35,7 +36,37 @@ public Name(String name) {
      * Returns true if a given string is a valid name.
      */
     public static boolean isValidName(String test) {
-        return test.matches(VALIDATION_REGEX);
+        return test.matches(VALIDATION_REGEX) && isFirstLetterCapitalised(test);
+    }
+
+    /**
+     * Validity check to ensure first letter is capitalised
+     * @param test The name to be tested.
+     * @return true if first letter for each name is capitalised.
+     */
+    private static boolean isFirstLetterCapitalised(String test) {
+        String[] name = test.split(" ");
+        return checkEachName(name);
+    }
+
+    /**
+     * Method helper to check for validity of name
+     * @param name An array containing each part of the name.
+     * @return True if name is valid.
+     */
+    private static boolean checkEachName(String[] name) {
+        for (int i = 0; i < name.length; i++) {
+            String word = name[i];
+            if (!Character.isUpperCase(word.charAt(0)) && !Character.isDigit(word.charAt(0))) {
+                return false;
+            }
+            for (int j = 1; j < word.length(); j++) {
+                if (!Character.isLowerCase(word.charAt(j)) && !Character.isDigit(word.charAt(j))) {
+                    return false;
+                }
+            }
+        }
+        return true;
```

Comment:
Very interesting method you have here. Just a suggestion for the future, you can avoid all these checks by modifying the equals method in the `Name` class from
```java 
fullName.equals(((Name) other).fullName)
```
to 
```java 
fullName.equalsIgnoreCase(((Name) other).fullName)
```
This way, the string clients `name` won't be restricted.

Reply:
Yup but a reason is bc changing equals method is changing behavior, but the intention here is only to add the missing validity check so HustleBook does not misbehave.